docs/recipes/vue.md

```diff
@@ -60,6 +80,71 @@ test('renders', t => {
 });
 ```
 
+## Sample data test
+
+```js
+import test from 'ava';
+import Vue from 'vue';
+import Component from 'component.vue';
+
+test('it has a message', t => {
+	const vm = new Vue(Component).$mount();
+	t.is('my message', vm.message);
+});
+```
+
+## Sample DOM testing with data manipulation
+
+When manipulating the data object of a Vue model, the DOM will not be updated immediately. It will be changed on the next tick.
+
+```js
+import test from 'ava';
+import Vue from 'vue';
+import Component from 'component.vue';
+
+test('see the updated message', t => {
+	const vm = new Vue(Component).$mount();
+	t.is(vm.$el.textContent, 'my message');
+
+	vm.message = 'my new message';
+	// this fails here: t.is('my new message', vm.$el.textContent)
+
+	Vue.nextTick(() => {
+		t.is(vm.$el.textContent, 'my new message');
+	});
+});
+```
+
+## Sample DOM testing with promises
+
+When using promises to fetch data to update the Vue model, you have to put your tests in the `then()` function of the pending promise.
+
+```js
+import test from 'ava';
+import Vue from 'vue';
+import Component from 'component.vue';
+
+// Don't forget to add the async prefix
+test('update the model using promises', async t => {
+	// assume the message data property is a pending promise defined in the Component.
+	const vm = new Vue(Component).$mount();
+
+	// just testing the data model can be achieved by using await
+	t.is(await vm.message, 'my message');
+
+	await vm.message.then((data) => {
```

Comment:
The `.then()` here is moot. You're already `await`'ing it. Just get value back the async/await way.